src/ipf.jl

```diff
@@ -51,23 +51,29 @@ function ipf(X::AbstractArray{<:Real}, mar::ArrayMargins; maxiter::Int = 1000, t
         throw(DimensionMismatch("The number of margins ($(ndims(mar))) needs to equal ndims(X) = $(ndims(X))."))
     end
     array_size = size(X)
-    if array_size != size(mar)
+    if array_size != size(mar) #calling size(mar) implicitly checks dimension sizes between arrays
         throw(DimensionMismatch("The size of the margins $(size(mar)) needs to equal size(X) = $array_size."))
     end
 
-    # margin consistency check
-    if !isconsistent(mar; tol = tol)
+    # margin consistency checks
+    if (!isconsistent(mar; tol = tol)) || !check_margin_totals(mar; tol = tol)
         # transform to proportions
-        @info "Inconsistent target margins, converting `X` and `mar` to proportions. Margin totals: $(sum.(mar.am))" 
+        @info "Inconsistent target margins, converting `X` and `mar` to proportions." 
         X /= sum(X)
         mar = proportion_transform(mar)
+
+        #recheck proportions across margins that appear more than once
+        if !check_margin_totals(mar; tol = tol)
+            throw(DimensionMismatch("Margin proportions inconsistent across dimensions"))
+        end
```

Comment:
Is this necessary? Is this not guaranteed? I understand why you'd want to double check, but maybe this is superfluous